apps/meteor/app/livechat/server/api/lib/livechat.ts

```diff
@@ -48,18 +55,22 @@ async function findDepartments(
 	}));
 }
 
-export function findGuest(token: string): Promise<ILivechatVisitor | null> {
-	return LivechatVisitors.getVisitorByToken(token, {
-		projection: {
-			name: 1,
-			username: 1,
-			token: 1,
-			visitorEmails: 1,
-			department: 1,
-			activity: 1,
-			contactId: 1,
-		},
-	});
+export function findGuest(token: string, sourceType?: OmnichannelSourceType): Promise<ILivechatVisitor | null> {
```

Comment:
It would be better to create a new function for filtering it by source type instead of reusing this one and making the filter optional...